gadget/update.go

```diff
@@ -1071,7 +1070,15 @@ func buildVolumeStructureToLocation(mod Model,
 	return volumeStructureToLocation, nil
 }
 
-// exposed for mocking later on
+func MockVolumeStructureToLocationMap(f func(_ GadgetData, _ Model, _ map[string]*LaidOutVolume) (map[string]map[int]StructureLocation, error)) (restore func()) {
+	old := volumeStructureToLocationMap
+	volumeStructureToLocationMap = f
+	return func() {
+		volumeStructureToLocationMap = old
+	}
```

Comment:
This is needed from the overlord tests which are outside of the gadget package so it has to be exported unfortunately

Reply:
hmm also is it kosher to import testutil into a production package like this? all the existing usages of testutil.Backup() I see are from `*_test.go` files so we are not importing testutil into the main package however since this is in a main package it will now be included everywhere and the implications of that are somewhat unclear

Reply:
we probably don't want to import testutil here. So for now we can't use the helper here.